subtrack.MAUI/Services/SubscriptionsCalculator.cs

```diff
@@ -65,14 +65,26 @@ public DateTime GetNextPaymentDate(Subscription subscription)
     {
         int startDay = subscription.FirstPaymentDay;
         var lastPayment = subscription.LastPayment;
-
-        var nextMonthDate = lastPayment.AddMonths(1);
-        var nextMonthTotalDays = DateTime.DaysInMonth(nextMonthDate.Year, nextMonthDate.Month);
-
-        if (startDay > nextMonthTotalDays)
-            return nextMonthDate;
-
-        return new DateTime(nextMonthDate.Year, nextMonthDate.Month, startDay);
+        switch (subscription.BillingOccurrence)
+        {
+            case BillingOccurrence.Week:
+                var nextWeekDate = lastPayment.AddDays(7*subscription.BillingInterval);
```

Comment:
Does .AddWeeks not work?

Reply:
It works but it requires setting up a calendar property and I though it was too much set up for that https://learn.microsoft.com/en-us/dotnet/api/system.globalization.calendar.addweeks?view=net-7.0
returnTime = DateTime.SpecifyKind(cal.AddWeeks(time, weeks), time.Kind);

Reply:
Aha, how about we add our own .AddWeeks method?

```
public static class DateExtensions
{
    public static DateTime AddWeeks(this DateTime date, int weeks)
    {
        return date.AddDays(weeks * 7);
    }
```